# ESCUELA POLITÉCNICA NACIONAL
# FACULTAD DE INGENIERÍA EN SISTEMAS
# MATERIA: MÉTODOS NUMÉRICOS
# NOMBRE: DANIEL ISMAEL FLORES ESPÍN

#### 1. Calcule los errores absoluto y relativo en las aproximaciones de p por p*


#### a) $p =\pi,  p*=\frac{22}{7}$ 

$E_a = |p - p^*| = |\pi - \frac{22}{7}|$

In [2]:
import math
abs1 = abs(math.pi-(22/7))
print(f"{abs1:.10f}")

0.0012644893


0.0012644893% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|\pi - \frac{22}{7}|}{|\pi|}$

In [7]:
rel1 = abs((math.pi-(22/7))/(math.pi))
print(f"{rel1:.10f}")

0.0004024994


0.04024994% de Error

#### b) $p =\pi,  p*=3.1416$ 

$E_a = |p - p^*| = |\pi - 3.1416|$

In [8]:
abs2 = abs(math.pi-3.1416)
print(f"{abs2:.10f}")

0.0000073464


0.00073464% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|\pi - 3.1416|}{|\pi|}$

In [17]:
rel2 = abs((math.pi-3.1416)/(math.pi))
print(f"{rel2:.10f}")

0.0000023384


 0.00023384% de Error

#### c) $p =e,  p*=2.718$ 8

$E_a = |p - p^*| = |e - 2.718|$

In [20]:
abs3 = abs(math.e-2.718)
print(f"{abs3:.10f}")

0.0002818285


 0.02818285% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|e - 2.718|}{|e|}$

In [21]:
rel3 = abs((math.e-2.718)/(math.e))
print(f"{rel3:.10f}")

0.0001036789


 0.01036789% de Error

#### d) $p =\sqrt{2},  p*=1.414$ 

$E_a = |p - p^*| = |\sqrt{2} - 1.414|$

In [31]:
abs4 = abs(math.sqrt(2)-1.414)
print(f"{abs4:.10f}")

0.0002135624


0.02135624% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|\sqrt{2} - 1.414|}{|\sqrt{2}|}$

In [34]:
rel4 = abs((math.sqrt(2)-1.414)/(math.sqrt(2)))
print(f"{rel4:.10f}")

0.0001510114


0.01510114% de Error

#### 2. Calcule los errores absoluto y relativo en las aproximaciones de p por p*

#### a) $p =e^{10},  p*=22000$ 

$E_a = |p - p^*| = |e^{10} - 22000|$

In [39]:
abs5 = abs(math.pow(math.e,10)-22000)
print(f"{abs5:.10f}")

26.4657948067


2646.57948067% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|e^{10} - 22000|}{|e^{10}|}$

In [42]:
rel5 = abs((math.pow(math.e,10)-22000)/(math.pow(math.e,10)))
print(f"{rel5:.10f}")

0.0012015452


0.0012015452% de Error

#### b) $p =10^{\pi},  p*=1400$ 

$E_a = |p - p^*| = |10^{\pi} - 1400|$

In [43]:
abs6 = abs(math.pow(10,math.pi)-1400)
print(f"{abs6:.10f}")

14.5442686330


1454.42686330% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|10^{\pi} - 1400|}{|10^{\pi}|}$

In [47]:
rel6 = abs((math.pow(10,math.pi)-1400)/(math.pow(10,math.pi)))
print(f"{rel6:.10f}")

0.0104978227


0.0104978227% de Error

#### c) $p =8!,  p*=39900$ 

$E_a = |p - p^*| = |8! - 39900|$

In [45]:
abs7 = abs(math.factorial(8)-39900)
print(f"{abs7:.10f}")

420.0000000000


42000.00000000% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|8! - 39900|}{|8!|}$

In [50]:
rel7 = abs((math.factorial(8)-39900)/(math.factorial(8)))
print(f"{rel7:.10f}")

0.0104166667


1.04166667% de Error

#### d) $p =9!,  p*=\sqrt{18\pi}(\frac{9}{e})^9$ 

$E_a = |p - p^*| = |9! - \sqrt{18\pi}(\frac{9}{e})^9|$

In [55]:
abs8 = abs(math.factorial(9)-(math.sqrt(18*math.pi)*math.pow((9/math.e),9)))
print(f"{abs8:.10f}")

3343.1271580516


334312.71580516% de Error

$E_r = \frac{|p - p^*|}{|p|} = \frac{|8! - 39900|}{|8!|}$

In [56]:
rel8 = abs((math.factorial(9)-(math.sqrt(18*math.pi)*math.pow((9/math.e),9)))/(math.factorial(9)))
print(f"{rel8:.10f}")

0.0092127622


0.92127622% de Error

#### 3. Encuentre el intervalo más largo en el que se debe encontrar 𝑝* para aproximarse a 𝑝 con error relativo máximo de  $10^{−4}$ para cada valor de 𝑝. 

$\text{Error relativo} = \frac{|p^* - p|}{|p|}$

$\frac{|p^* - p|}{|p|} \leq 10^{-4}$

$p - 10^{-4} \cdot p \leq p^* \leq p + 10^{-4} \cdot p$

#### a) ${\pi}$

$p = \pi$

$\pi - 10^{-4} \cdot \pi \leq p^* \leq \pi + 10^{-4} \cdot \pi$

$\pi - 3.1415926536 \cdot 10^{-4} \leq p^* \leq \pi + 3.1415926536 \cdot 10^{-4}$

$3.1412784943 \leq p^* \leq 3.1419068129$

#### b)$e$

$p = e$

$e - 10^{-4} \cdot e \leq p^* \leq e + 10^{-4} \cdot e$

$2.7182818285 - 2.7182818285 \cdot 10^{-4} \leq p^* \leq 2.7182818285 + 2.7182818285 \cdot 10^{-4}$

$2.7180090000 \leq p^* \leq 2.7185546567$

#### c) $\sqrt{2}$

$\sqrt{2} - 10^{-4} \cdot \sqrt{2} \leq p^* \leq \sqrt{2} + 10^{-4} \cdot \sqrt{2}$

$1.4142135624 - 1.4142135624 \cdot 10^{-4} \leq p^* \leq 1.4142135624 + 1.4142135624 \cdot 10^{-4}$

$1.4140721411 \leq p^* \leq 1.4143549836$

#### d) $\sqrt[3]{7}$

$\sqrt{7} - 10^{-4} \cdot \sqrt{7} \leq p^* \leq \sqrt{7} + 10^{-4} \cdot \sqrt{7}$

$2.6457513111 - 2.6457513111 \cdot 10^{-4} \leq p^* \leq 2.6457513111 + 2.6457513111 \cdot 10^{-4}$

$2.6454867360 \leq p^* \leq 2.6460158862$

#### 4. Use la aritmética de redondeo de tres dígitos para realizar lo siguiente. Calcule los errores absoluto y relativo 
con el valor exacto determinado para por lo menos cinco dígitos.

#### a) $\frac{\frac{13}{14}-\frac{5}{7}}{2e-5.4}$

$\frac{13}{14} = 0.9285714286$ → 0.929

$\frac{5}{7} = 0.7142857143$ → 0.714

$\frac{13}{14}-\frac{5}{7} = 0.9285714286−0.7142857143=0.2142857143$ → 0.929 − 0.714=0.215

$2e−5.4=2×2.71828−5.4=5.43656−5.4=0.03656$

$2e=2×2.72=5.44
5.44−5.4=0.04$

$\frac{0.215}{0.04} =5.375$

$E 
a
​
 =∣Valor Aproximado−Valor Exacto∣=∣5.375−5.86247∣=0.48747$

$E_r = \frac{0.48747}{5.86247} = 0.08319$

#### b) $-10\pi+6e-\frac{3}{61}$

$-10\pi = -10 \times 3.1415926536 = -31.4159265360$

$6e = 6 \times 2.7182818285 = 16.3096909710$

$\frac{3}{61} = 0.0491803279$

$-31.4159265360 + 16.3096909710 - 0.0491803279 = -15.1554158929$

$-15.2$

$|-15.1554158929 - (-15.2)| = 0.0445841071$

$\frac{0.0445841071}{15.1554158929} = 2.9414620453 \times 10^{-3}$

c) $(\frac{2}{9})(\frac{9}{11})$

$\frac{2}{9} = 0.2222222222$

$\frac{9}{11} = 0.8181818182$

$0.2222222222 \times 0.8181818182 = 0.1818181818$

$0.182$

$|0.1818181818 - 0.182| = 0.0001818182$

$\frac{0.0001818182}{0.1818181818} = 9.9999999991 \times 10^{-4}$

d) $\frac{\sqrt{13}+\sqrt{11}}{\sqrt{13}-\sqrt{11}}$

$\sqrt{13} = 3.6055512755$

$\sqrt{11} = 3.3166247904$

$3.6055512755 + 3.3166247904 = 6.9221760659$

$3.6055512755 - 3.3166247904 = 0.2889264851$

$\frac{6.9221760659}{0.2889264851} = 23.9600672727$

$24.0$

$|23.9600672727 - 24.0| = 0.0399327273$

$\frac{0.0399327273}{23.9600672727} = 1.6668306113 \times 10^{-3}$

#### 5.Los primeros tres términos diferentes a cero de la serie de Maclaurin para la función arcotangente son:
$x-(\frac{1}{3}) x^{3} + (\frac{1}{5}) x^{5}$ 
Calcule los errores absoluto y relativo en las siguientes aproximaciones de $\pi$ mediante el polinomio en lugar del arcotangente:

a. $4[arctan(\frac{1}{2})+arctan(\frac{1}{3})]$

$\arctan(x) = x - \frac{1}{3}x^3 + \frac{1}{5}x^5 + \dots$

$\arctan\left( \frac{1}{2} \right) \approx \frac{1}{2} - \frac{1}{3} \left( \frac{1}{2} \right)^3 + \frac{1}{5} \left( \frac{1}{2} \right)^5$

$\arctan\left( \frac{1}{2} \right) \approx 0.5000000000 - 0.0416666667 + 0.0062500000 = 0.4645833333$

$\arctan\left( \frac{1}{3} \right) \approx \frac{1}{3} - \frac{1}{3} \left( \frac{1}{3} \right)^3 + \frac{1}{5} \left( \frac{1}{3} \right)^5$

$\arctan\left( \frac{1}{3} \right) \approx 0.3333333333 - 0.0123456790 + 0.0008230453 = 0.3218106996$

$\arctan\left( \frac{1}{2} \right) + \arctan\left( \frac{1}{3} \right) = 0.4645833333 + 0.3218106996 = 0.7863940329$

$4 \times 0.7863940329 = 3.1455761316$

ERROR:

$\pi = 3.1415926536$

$|3.1455761316 - 3.1415926536| = 0.0039834780$

$\frac{0.0039834780}{3.1415926536} = 1.2681963481 \times 10^{-3}$

b. $16 acrtan(\frac{1}{5}) - 4 arctan(\frac{1}{239})$

$\arctan\left( \frac{1}{5} \right) \approx \frac{1}{5} - \frac{1}{3} \left( \frac{1}{5} \right)^3 + \frac{1}{5} \left( \frac{1}{5} \right)^5$

$\arctan\left( \frac{1}{5} \right) \approx 0.2000000000 - 0.0013333333 + 0.0000320000 = 0.1986986667$

$\arctan\left( \frac{1}{239} \right) \approx \frac{1}{239} - \frac{1}{3} \left( \frac{1}{239} \right)^3 + \frac{1}{5} \left( \frac{1}{239} \right)^5$

$\arctan\left( \frac{1}{239} \right) \approx 0.0041841004 - 1.0310096 \times 10^{-7} + 2.4373424 \times 10^{-12} = 0.0041840003$

$16 \times 0.1986986667 = 3.1791786667$

$4 \times 0.0041840003 = 0.0167360012$

$3.1791786667 - 0.0167360012 = 3.1624426655$

ERROR:

$|3.1624426655 - 3.1415926536| = 0.0208500119$

$\frac{0.0208500119}{3.1415926536} = 6.6372267515 \times 10^{-3}$

#### 6. El número $e$ se puede definir por medio de $\displaystyle\sum_{n=0}^\infty (\frac{1}{n!})$  , donde $ 𝑛!n n(𝑛 − 1) ⋅⋅⋅ 2 $⋅ 1 pa$n 𝑛 $≠  $ y
0!$ = 1. Calcule los errores absoluto y relativo en la siguiente aproximació $e$ de 

a. $\displaystyle\sum_{n=0}^{5} (\frac{1}{n!})$

$e = \sum_{n=0}^{\infty} \frac{1}{n!}$

$\frac{1}{0!} = 1$

$\frac{1}{1!} = 1$

$\frac{1}{2!} = 0.5$

$\frac{1}{3!} = 0.1666666667$

$1 + 1 + 0.5 + 0.1666666667 + 0.0416666667 + 0.0083333333 = 2.7183333333$

$e = 2.7182818285$

$|2.7183333333 - 2.7182818285| = 0.0000515048$

$\frac{0.0000515048}{2.7182818285} = 1.8950519840 \times 10^{-5}$

b. $\displaystyle\sum_{n=0}^{10} (\frac{1}{n!})$

$\frac{1}{6!} = 0.0013888889$

$\frac{1}{7!} = 0.0001984127$

$\frac{1}{8!} = 0.0000248016$

$\frac{1}{9!} = 0.0000027557$

$\frac{1}{10!} = 0.0000002756$

$2.7183333333 + 0.0013888889 + 0.0001984127 + 0.0000248016 + 0.0000027557 + 0.0000002756 = 2.7182818011$

$|2.7182818011 - 2.7182818285| = 0.0000000274$

$\frac{0.0000000274}{2.7182818285} = 1.0073743100 \times 10^{-8}$

#### 7. Suponga que dos puntos ($x_0$,$y_0$ se encuentran en línea racta con $y_1 \neq y_0$. Existen dos fórmulas para encontrar la intersección x de la línea:
$x = \frac{x_0y_1-x_1y_0}{y_1-y_0}$ y $x=x_0-\frac{(x_1-x_0)y_0}{y_1-y_0}$

a. Use los datos $(x_0,y_0)$=(1.31,3.24) y $(x_1,y_1)$ = (1.93,5.76) y la aritmética de redondeo de tres dígitos para calcular la intersección con x de ambas manera ¿Cuál método es mejor y por qué?

$(x_0, y_0) = (1.31, 3.24) \quad \text{y} \quad (x_1, y_1) = (1.93, 5.76)$

$x = \frac{x_0 y_1 - x_1 y_0}{y_1 - y_0} \quad \text{y} \quad x = x_0 - \frac{(x_1 - x_0) y_0}{y_1 - y_0}$

$x = \frac{x_0 y_1 - x_1 y_0}{y_1 - y_0}$

$x_0 y_1 = 1.31 \times 5.76 = 7.5456 \quad \text{(redondeado: 7.55)}$

$x_1 y_0 = 1.93 \times 3.24 = 6.2532 \quad \text{(redondeado: 6.25)}$

$x_0 y_1 - x_1 y_0 = 7.55 - 6.25 = 1.30$

$y_1 - y_0 = 5.76 - 3.24 = 2.52$

$x = \frac{1.30}{2.52} = 0.5159 \quad \text{(redondeado: 0.516)}$

Segunda Fórmula:

$x = x_0 - \frac{(x_1 - x_0) y_0}{y_1 - y_0}$

$x_1 - x_0 = 1.93 - 1.31 = 0.62$

$0.62 \times 3.24 = 2.0088 \quad \text{(redondeado: 2.01)}$

$\frac{2.01}{2.52} = 0.7976 \quad \text{(redondeado: 0.798)}$

$x = 1.31 - 0.798 = 0.512$

### ¿Cuál método es mejor y por qué?

- La primera fórmula es más directa y tiende a ser más precisa cuando los valores de \(x_0\) y \(x_1\) son pequeños, ya que evita multiplicaciones adicionales con \(y_0\).
- La segunda fórmula puede introducir más error de redondeo porque involucra más operaciones intermedias.

Por lo tanto, en este caso, la primera fórmula es preferible porque es más sencilla y menos propensa a errores acumulativos en los pasos de cálculo.